core/components/atoms/input/Input.tsx

```diff
@@ -257,9 +259,9 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>((props, forw
           <Text appearance="subtle">{inlineLabel}</Text>
         </div>
       )}
-      {size !== 'tiny' && icon && (
+      {icon && (
         <div className={leftIconClass}>
-          <Icon name={icon} size={sizeMapping[size]} type={iconType} />
+          <Icon name={icon} size={size === 'tiny' ? 14 : sizeMapping[size]} type={iconType} />
```

Comment:
Please move this size calculation logic into some variable and then use that variable here